command/v7/unbind_service_command.go

```diff
@@ -0,0 +1,88 @@
+package v7
+
+import (
+	"code.cloudfoundry.org/cli/actor/actionerror"
+	"code.cloudfoundry.org/cli/actor/v7action"
+	"code.cloudfoundry.org/cli/command/flag"
+	"code.cloudfoundry.org/cli/command/v7/shared"
+)
+
+type UnbindServiceCommand struct {
+	BaseCommand
+
+	RequiredArgs    flag.BindServiceArgs `positional-args:"yes"`
+	Wait            bool                 `short:"w" long:"wait" description:"Wait for the bind operation to complete"`
+	relatedCommands interface{}          `related_commands:"apps, delete-service, services"`
+}
+
+func (cmd UnbindServiceCommand) Execute(args []string) error {
+	if err := cmd.SharedActor.CheckTarget(true, true); err != nil {
+		return err
+	}
+
+	if err := cmd.displayIntro(); err != nil {
+		return err
+	}
+
+	stream, warnings, err := cmd.Actor.DeleteServiceAppBinding(v7action.DeleteServiceAppBindingParams{
+		SpaceGUID:           cmd.Config.TargetedSpace().GUID,
+		ServiceInstanceName: cmd.RequiredArgs.ServiceInstanceName,
+		AppName:             cmd.RequiredArgs.AppName,
+	})
+	cmd.UI.DisplayWarnings(warnings)
+	switch err.(type) {
+	case nil:
+	case actionerror.ServiceBindingNotFoundError:
+		cmd.UI.DisplayText("Binding between {{.ServiceInstanceName}} and {{.AppName}} did not exist", cmd.names())
+		cmd.UI.DisplayOK()
+		return nil
+	default:
+		return err
+	}
+
+	completed, err := shared.WaitForResult(stream, cmd.UI, cmd.Wait)
+	switch {
```

Comment:
I'm particularly not a big fan of this type of `switch`. I'm just used to believe that all cases are assertions on the same "variable" (if that makes sense). Not advocating to change it though, since it seems to be [idiomatic](https://golang.org/doc/effective_go.html#switch) to use `switch` this way 🤔